src/PIL/Image.py

```diff
@@ -2116,25 +2243,31 @@ def transform(self, size, method, data=None, resample=NEAREST,
 
         if hasattr(method, "getdata"):
             # compatibility w. old-style transform objects
-            method, data = method.getdata()
+            # ignoring for typing, this is old compatibility
+            method, data = method.getdata() # type: ignore
 
         if data is None:
             raise ValueError("missing method data")
 
         im = new(self.mode, size, fillcolor)
         if method == MESH:
             # list of quads
-            for box, quad in data:
+            data_mesh = None # type: List[Tuple[LURD,LURD]]
+            data_mesh = data # type: ignore
```

Comment:
What's going on here, setting `data_mesh` twice?

Reply:
Yes, as I mentioned above. If you wish to maintain the previous comment of what it "should" be, ideally, then perhaps just put it as the 'post-ignore' comment?

Reply:
Basically, the first sets the type, and the second actually sets the variable, but we're forcing the typechecker to accept that we know that data is the correct type. This sort of idiom is required(?) when we're unpacking a datastructure that's goat a bunch of type overloads.